### PR TITLE
Fix some realloc error handling issues.

### PR DESCRIPTION
**Pull request by bernd-edlinger**
Fix a few realloc error handling issues that were reported in #2496.

### CODE REVIEW

**Comment by bernd-edlinger** [comment]
This PR is against the 1.0.2 branch.

File: ssl/t1_ext.c [hunk 1]
```diff
@@ -223,16 +223,14 @@ static int custom_ext_meth_add(custom_ext_methods *exts,
...
-                                  (exts->meths_count +
-                                   1) * sizeof(custom_ext_method));
-
-    if (!exts->meths) {
-        exts->meths_count = 0;
+    meth = OPENSSL_realloc(exts->meths,
+                           (exts->meths_count + 1)
+                           * sizeof(custom_ext_method));
+    if (meth == NULL)
         return 0;
```
Comment (mattcaswell) [inline ssl/t1_ext.c]: Behaviour on failure should probably be consistent with the master/1.1.0 branch

Comment (bernd-edlinger) [inline ssl/t1_ext.c]: What a strange behaviour...

Comment (mattcaswell) [inline ssl/t1_ext.c]: I didn't say which was right! Just that they should be consistent :-)

Comment (bernd-edlinger) [inline ssl/t1_ext.c]: Ah, yeah :-)
At least the documentation does not mention that at all.
Which behaviour looks more reasonable for you?

Comment (mattcaswell) [inline ssl/t1_ext.c]: Well the behaviour you have here looks more sensible

**Comment by richsalz** [comment]
b75dbf3 in 1.0.2;  thanks!

**Pull request closed without merging**